browser/ui/webui/brave_web_ui_controller_factory.cc

```diff
@@ -90,7 +92,12 @@ WebUIController* NewWebUI(WebUI* web_ui, const GURL& url) {
   auto host = url.host_piece();
   Profile* profile = Profile::FromBrowserContext(
       web_ui->GetWebContents()->GetBrowserContext());
-  if (host == kSkusInternalsHost) {
+  if (host == kAdsInternalsHost) {
+    return new AdsInternalsUI(
+        web_ui, url.host(),
+        brave_ads::AdsServiceFactory::GetForProfile(profile),
+        profile->GetPrefs());
+  } else if (host == kSkusInternalsHost) {
```

Comment:
It can be done on a follow up, but webui handling in being migrated to use `DefaultWebUIConfig`. For reference. https://github.com/brave/brave-core/pull/25808.